daemon/probes.go

```
@@ -0,0 +1,160 @@
+		"/proc/config.gz",
+	}
+	localConfigLocationsGz := []string{
+		fmt.Sprintf("/boot/config-%s", getKernelVersionStr()),
+	}
+
+	for _, localConfigLocation := range localConfigLocations {
+		if _, err := os.Stat(localConfigLocation); !os.IsNotExist(err) {
+			fConfig, err := os.Open(localConfigLocation)
+			if err != nil {
+				return nil, err
+			}
+			return ioutil.ReadAll(fConfig)
+		}
+	}
+
+	for _, localConfigLocation := range localConfigLocationsGz {
+		if _, err := os.Stat(localConfigLocation); !os.IsNotExist(err) {
+			fConfig, err := os.Open(localConfigLocation)
+			if err != nil {
+				return nil, err
+			}
+			gConfig, err := gzip.NewReader(fConfig)
+			if err != nil {
+				return nil, err
+			}
+			return ioutil.ReadAll(gConfig)
+		}
+	}
+
+	return nil, fmt.Errorf("could not find kernel config file")
+}
+
+func probeKernelConfig() error {
+	requiredParams := []string{
+		"CONFIG_BPF=y",
+		"CONFIG_BPF_SYSCALL=y",
+		"CONFIG_NET_SCH_INGRESS=(m|y)",
+		"CONFIG_NET_CLS_BPF=(m|y)",
+		"CONFIG_NET_CLS_ACT=y",
+		"CONFIG_BPF_JIT=y",
+		"CONFIG_HAVE_EBPF_JIT=y",
+	}
+
+	kConfig, err := kernelConfig()
+	if err != nil {
+		return err
+	}
+
+	for _, param := range requiredParams {
+		re := regexp.MustCompile(param)
+		if !re.Match(kConfig) {
+			return fmt.Errorf("%s is not in kernel configuration", param)
+		}
+	}
+
+	return nil
+}
+
+func probeRunTc() error {
+	return nil
+}
+
+func copyFile(src, dest string) error {
+	in, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+	defer in.Close()
+
+	out, err := os.Create(dest)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+
+	_, err = io.Copy(out, in)
+	return err
+}
+
+func probeRunLl(probesDir, libIncludeDir, outDir string) error {
+	outFilename := path.Join(outDir, "raw_probe.t")
+
+	err := filepath.Walk(probesDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if strings.HasSuffix(info.Name(), ".t") {
+			copyFile(path, outFilename)
```

You are ignoring the error returned by this function.